cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -69,11 +70,12 @@ import Language.Haskell.Extension (Language(..))
 createProject
     :: Interactive m
     => Verbosity
+    -> Compiler
     -> InstalledPackageIndex
     -> SourcePackageDb
     -> InitFlags
     -> m ProjectSettings
-createProject v pkgIx srcDb initFlags = do
+createProject v _comp pkgIx srcDb initFlags = do
```

Comment:
This is unused in 2 cases (interactive + simple). I'd just configure `NonInteractive.createProject` to take the compiler arg first and then return `NonInteractive.createProject comp` as the function returned in the where clause.